### ISSUE TITLE
Document exceptions

### ISSUE THREAD

**Issue by akihironitta** [Lightning-AI/lightning#5434]
## 📚 Documentation
This is only a suggestion, but I find it useful to have what exceptions functions/classes can raise in the docs, so I'd like to suggest adding the `Raises:` section to public functions/classes.

**Progress**
`dirname $(grep -rl "raise " pytorch_lightning)|sort|uniq|awk '{print "- [ ] "$1}'`

- [ ] pytorch_lightning/accelerators (#9558)
- [ ] pytorch_lightning/callbacks
- [ ] pytorch_lightning/callbacks/progress
- [ ] pytorch_lightning/core
- [ ] pytorch_lightning/core/mixins
- [ ] pytorch_lightning/loggers
- [ ] pytorch_lightning/loops
- [ ] pytorch_lightning/loops/dataloader
- [ ] pytorch_lightning/loops/epoch
- [ ] pytorch_lightning/loops/optimization
- [ ] pytorch_lightning/overrides
- [ ] pytorch_lightning/plugins
- [ ] pytorch_lightning/plugins/environments
- [ ] pytorch_lightning/plugins/io
- [ ] pytorch_lightning/plugins/precision
- [ ] pytorch_lightning/plugins/training_type
- [ ] pytorch_lightning/profiler
- [ ] pytorch_lightning/trainer
- [ ] pytorch_lightning/trainer/connectors
- [ ] pytorch_lightning/trainer/connectors/logger_connector
- [ ] pytorch_lightning/tuner
- [ ] pytorch_lightning/utilities

**Comment by akihironitta** [comment]
The following docstring will look like the below image:
```python
    """
    Raises:
        MisconfigurationException:
            If ``logging_interval`` is none of ``"step"``, ``"epoch"``, and ``None``.
    """
```
![tmp](https://user-images.githubusercontent.com/20610905/104081149-207fe700-5270-11eb-930f-5fddf0e0b98b.png)

**Comment by tchaton** [comment]
Hey @akihironitta,

Great idea. Let's start adding them.

Best,
T.C

**Comment by akihironitta** [comment]
Could you add `"good first issue"` label to this issue so that new contributors can work on this issue?

**Comment by kaushikb11** [comment]
@akihironitta Done! :)

**Comment by dipam7** [comment]
I am new to PyTorch lightning and I can work on this issue!

**Comment by akihironitta** [comment]
@dipam7 Hi, thank you for your contribution! To avoid conflicts with other contributors, could I ask which submodule you'd like to work on?

**Comment by AlKun25** [comment]
@akihironitta Please have a look at my PR #6171, and if there are any exceptions I have missed.

**Comment by dipam7** [comment]
@akihironitta Hey, I'm currently looking at metrics/classification. I think that would be the easiest for me to do. I'll comment here when I move on to other submodules.

**Comment by dipam7** [comment]
@akihironitta Just submitted a PR #6190 if you can take a look

**Comment by prajakta0111** [comment]
@akihironitta @dipam7 I am submitting a PR for metrics/regression

**Comment by AlKun25** [comment]
@akihironitta  There doesn't seem to be any *errors* or *exceptions* in ``overrides`` folder

**Comment by dipam7** [comment]
@akihironitta I'm doing metrics/functional as well

**Comment by AlKun25** [comment]
@akihironitta There are no changes required for ``pytorch_lightning/distributed/*``. And if ``cluster_environments`` is the file ``pytorch_lightning/plugins/environments/cluster_environment.py``, then if that doesn't require any changes. I couldn't find any folder named `cluster_environments`

**Comment by akihironitta** [comment]
@AlKun25 Thank you for letting me know!

**Comment by dipam7** [comment]
@akihironitta Just submitted the PR for functional. #6273

**Comment by djthegr8** [comment]
If it is not conflicting with someone else, I could work on `/utilities*`

**Comment by akihironitta** [comment]
Hi @djthegr8, that would be great!

**Comment by ananthsub** [comment]
@Borda should we also add this (and type hints) to the new contributor guide for what to do when submitting PRs?

**Comment by dingusagar** [comment]
I can do this for pytorch_lightning/trainer if no one is already working on this.

**Comment by dingusagar** [comment]
hi @akihironitta, could you please check my PR #16684
